### PR TITLE
Authentication handling - JWT token retrieval and refreshment

### PR DESCRIPTION
**Pull request by krishnaKapadia**
This PR focuses on handling JWT authentication to ensure that subsequent requests can be made to the Public API securely.

Changes include:
- Adding a `TokenManager` to handle retrieval & persisting the token across multiple browser loads.
- Ensure that local storage exists before attempting to add/retrieve from it.
- Functionality to retrieve a new token if the current one expires or isn't valid.

### CODE REVIEW

**Comment by krishnaKapadia** [comment]
This PR additionally adds support for all current public api endpoints related to the `Alert` entity.

File: lib/auth/tokenManager.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import jwtDecode from "jwt-decode";
+
+import { JwtToken } from "./models";
+import { IAuthStrategy } from ".";
+
+export class TokenManager {
+  private authStrategy: IAuthStrategy;
+  private token: string | undefined;
+  private saveToBrowserStorage: boolean;
+  private tokenStorageKey: string = "x-raygnapi";
```
Comment (MattByers) [inline lib/auth/tokenManager.ts]: Looks to be a typo on this key `x-raygnapi`

File: lib/entities/alerts/index.ts [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  /**
+   * Retrieves a list of alerts
+   * @param planIdentifier - Identifier of the target plan
+   * @param subscribedOnly - If true, only subscribed alerts will be returned
+   * @param page - The page number to return
+   * @param pageSize - The number of items per page. Max is 100
+   * @returns An array of {AlertSummary} objects
+   */
+  async getAll(planIdentifier: string, subscribedOnly: boolean = false, page: number = 1, pageSize = 20): Promise<IPagedEntity<IAlertSummary> | null> {
+    const urlSegments = ["user", GlobalConfig.userIdentifier, this.baseUrl, planIdentifier];
```
Comment (MattByers) [inline lib/entities/alerts/index.ts:43]: We will need to change this when the get alerts endpoint is updated to not pass through the user identifier. I'm moving onto that work next, so we can sync up and change it then.

**Review by MattByers — APPROVED** [review]
Looks good, just a couple of minor comments 👍

File: lib/entities/alerts/index.ts (hunk 2 above)
Comment (krishnaKapadia) [reply lib/entities/alerts/index.ts:43]: Ah yep, no worries will be a simple change 👍

**Pull request merged**